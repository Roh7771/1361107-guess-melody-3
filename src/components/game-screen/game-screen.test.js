import React from "react";
import renderer from "react-test-renderer";
import GameScreen from "./game-screen.jsx";

const mock = {
  type: `artist`,
  children: <p>What a wonderfull day</p>,
};

it(`with type GameType.ARTIST`, () => {
  const {type, children} = mock;
  const tree = renderer.create(
      <GameScreen
        type={type}
      >
        {children}
      </GameScreen>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
