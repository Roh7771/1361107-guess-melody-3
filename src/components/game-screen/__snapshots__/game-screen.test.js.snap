// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with type GameType.ARTIST 1`] = `
<main
  className="app"
>
  <svg
    style={
      Object {
        "left": "-1200em",
        "position": "absolute",
      }
    }
    xmlns="http://www.w3.org/2000/svg"
  >
    <filter
      id="blur"
    >
      <feGaussianBlur
        in="SourceGraphic"
        stdDeviation="5"
      />
      <feOffset
        dx="0"
        dy="0"
      />
      <feMerge>
        <feMergeNode />
        <feMergeNode
          in="SourceGraphic"
        />
      </feMerge>
    </filter>
  </svg>
  <section
    className="main"
    id="root"
  >
    <section
      className="game game--artist}"
    >
      <header
        className="game__header"
      >
        <a
          className="game__back"
          href="#"
        >
          <span
            className="visually-hidden"
          >
            Сыграть ещё раз
          </span>
          <img
            alt="Угадай мелодию"
            className="game__logo"
            src="img/melody-logo-ginger.png"
          />
        </a>
        <svg
          className="timer"
          viewBox="0 0 780 780"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            className="timer__line"
            cx="390"
            cy="390"
            r="370"
            style={
              Object {
                "filter": "url(#blur)",
                "transform": "rotate(-90deg) scaleY(-1)",
                "transformOrigin": "center",
              }
            }
          />
        </svg>
        <div
          className="game__mistakes"
        >
          <div
            className="wrong"
          />
          <div
            className="wrong"
          />
          <div
            className="wrong"
          />
        </div>
      </header>
      <p>
        What a wonderfull day
      </p>
    </section>
  </section>
</main>
`;
